### PR TITLE
CV-317: Support rectangular input for AHOY export

### PR DESCRIPTION
**Pull request by Zwicii**
## What?
 
Support rectangular input shapes when exporting an AHOY model.

## Why?
 
Currently, just square input shapes are used in our products, but this will change in the future. This changes enable support for rectangular input shapes.

## How?

`imgsz` can now be a `tuple` of `(height, width)`, allowing for rectangular input shapes instead of just supporting square dimensions.

Previously, `imgsz` was treated as a single value and used to create a square dummy input tensor when exporting the model. Now, the export process correctly supports rectangular input shapes by handling `imgsz ` as a `(height, width)` tuple.

#### Before
Before, `imgsz` was used to create a dummy input tensor, assuming it was always square:

```python
# Create dummy input  
image = torch.zeros((batch_size, 3, imgsz, imgsz), device=model.device).byte()  # B, C, H, W  
```

#### Now
Now, `imgsz` is correctly interpreted as `(height, width)`, allowing for non-square inputs:

```python
# Create dummy input  
image = torch.zeros((batch_size, 3, imgsz[0], imgsz[1]), device=model.device).byte()  # B, C, H, W  
```

#### `imgsz` Helper Function
A helper function `_transform_imgsz` was introduced to ensure robust handling of input sizes:

- If int: converted to (imgsz, imgsz)
- If list/tuple with one element: converted to (element, element)
-  If list/tuple with two elements: converted to (element1, element2)

#### Updated Naming Convention

The exported file name now follows this format:
`ahoy_{bz}_sz{h}x{w}.onnx`

## Backout plan

Revert the changes
 
## Testing
AHOY model was exported.

image

Input size is really rectangular.
image

Inference is still working :)
image

### CODE REVIEW

File: export_ahoy.py [hunk 1]
```diff
@@ -74,19 +76,52 @@ def get_weights_path(weights_path: str) -> str:
...
+        return (imgsz, imgsz)
+    
+    # Handle sequence cases
+    if isinstance(imgsz, (list, tuple)):
+        if len(imgsz) == 1:
+            return (imgsz[0], imgsz[0])
+        elif len(imgsz) == 2:
+            return (imgsz[0], imgsz[1])
+        else:
+            raise ValueError(f"Input size must have 1 or 2 elements, got {len(imgsz)}")
```
Comment (kevinconka) [inline export_ahoy.py]: there should always be a return value. In this case, if imgsz is a `str` then it won't satisfy and of the `if` conditions and return `None` implicitly.

We can reduce the logic to this, taking advantage of python's negative indices.

```python
if isinstance(imgsz, int):
    return imgsz, imgsz
if not isinstance(imgsz, (list, tuple)) or len(imgsz) not in (1, 2):
    raise ValueError(f"imgsz must be int or a list/tuple of 1 or 2 elements, got {imgsz}")
return imgsz[0], imgsz[-1]
```

File: export_ahoy.py [hunk 2]
```diff
@@ -74,19 +76,52 @@ def get_weights_path(weights_path: str) -> str:
...
 
 def main(
     det_weights: str,
     hor_weights: str,
-    imgsz: int,
+    imgsz: Tuple[int, int],
```
Comment (kevinconka) [inline export_ahoy.py]: Let's add a `Union`?

**Review by kevinconka — CHANGES_REQUESTED** [review]
Minor changes :)

File: export_ahoy.py (hunk 2 above)
Comment (Zwicii) [reply export_ahoy.py]: Yes forgot it thanks :)

File: export_ahoy.py (hunk 1 above)
Comment (Zwicii) [reply export_ahoy.py]: way better thanks :)

**Review by kevinconka — APPROVED** [review]
